### PR TITLE
Dropping 1% of cells and remaking UMAP

### PR DESCRIPTION
**Pull request by Nathaniel-github**

### CODE REVIEW

**Review by aarmey — COMMENTED** [review]
1. Does this work? Are you seeing the same weird UMAP issue as before? If so, are we sure that the cells are returned in the same order from `gateThomsonCells()` and the data importer? The gating function does a roundtrip through tensorFy().
2. UMAP is not quantitative and has no way for comparing whether two results are similar. We really should just focus on similarity among the factors (so long as you know you're eliminating cell types correctly).

File: sccp/figures/figureThomsonDrop.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    data.obs["cell_type"] = gateThomsonCells()
+
+    ax, f = getSetup((10,7), (1, 1))
+    subplotLabel(ax)
+
+    sampled_data = data.copy()
+    for drug in data.obs.Drugs.unique():
+        for cell_type in data.obs.cell_type.unique():
+            filt = (sampled_data.obs.cell_type != cell_type) | (
+                    sampled_data.obs.Drugs != drug
```
Comment (aarmey) [inline sccp/figures/figureThomsonDrop.py:34]: Why do you have to loop over drugs, if you build in the drug and cell type filtering as a logical or?

Comment (Nathaniel-github) [reply sccp/figures/figureThomsonDrop.py:34]: This one was a filter to just drop an equal 1% across the data to just pretend if those cells were never in the data in the first place

**Comment by Nathaniel-github** [comment]
1. Yes I'm still seeing the same results when dropping cells and even when not dropping cells my projections don't come out the same as Andrew's and I haven't been able to figure out why. So if I were to change that 0.99 to 1 it still plots a UMAP slightly different, was still discussing with Andrew about debugging this and we wanted to edit the gating function like you mentioned first to completely verify it is being assigned correctly.
2. For sure, I'll take a closer look at the factors when isolating a condition to drop and discuss with Andrew

**Comment by aarmey** [comment]
@Nathaniel-github we have greatly simplified how you pass around datasets. I updated your code as an example. In the process of going through this, we found an error in my handling of the factorization process. I think you'll find that the cell types now plot in a reasonable way.

**Comment by Nathaniel-asdrp** [comment]
@aarmey yes I just checked it out it looks much prettier 🤩. I'll check out the new functions and start using them

**Comment by Nathaniel-github** [comment]
@andrewram4287

**Comment by andrewram4287** [comment]
@Nathaniel-github We should start based on the import_thomson function to prevent issues overwriting the Pf2 AnnData object; plus we will be changing what cell types and drugs we want to remove so I think this route will end up easier

Right now the gateThomsonCells() doesn't fit the size of the import data so I'm currently investigating this issue

**Comment by aarmey** [comment]
@Nathaniel-github this looks great—I see one component with depletion in CTRL 4 and several marker genes of T cells. Merging so that we can work off of this in the future.

**Pull request merged**